inbm/dispatcher-agent/dispatcher/sota/snapshot.py

```diff
@@ -223,10 +225,13 @@ def recover(
         dispatcher_state.clear_dispatcher_state()
         if self.snap_num:
             self._rollback_and_delete_snap()
+            logger.debug("Rebooting to recover from failed SOTA...")
+            rebooter.reboot()
         else:
-            time.sleep(time_to_wait_before_reboot)
-        logger.debug("Rebooting to recover from failed SOTA...")
-        rebooter.reboot()
+            time.sleep(time_to_wait_before_reboot)            
+            if rebooter.is_reboot(self._device_reboot):    
+                logger.debug("Rebooting to recover from failed SOTA...")
+                rebooter.reboot()
```

Comment:
**Suggestion:** The recover method in DebianBasedSnapshot class should not attempt to reboot if _device_reboot is set to a value that indicates rebooting is not desired. The current implementation may lead to an unintended reboot. The condition to check rebooter.is_reboot(self._device_reboot) should be moved outside of the else block to ensure it's always evaluated. [bug]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        if self.snap_num:
            self._rollback_and_delete_snap()
        time.sleep(time_to_wait_before_reboot)
        if rebooter.is_reboot(self._device_reboot):    
            logger.debug("Rebooting to recover from failed SOTA...")
            rebooter.reboot()
```